File: CPP/run_time_polymorphisum.cpp
#include<iostream>
using namespace std;
class shape
{
	public:
		virtual void draw()
		{
			cout<<"Draw method of Shape class"<<endl;
		}
};
class rectangle:public shape
{
	public:
		void draw()
		{
			cout<<"Draw method of Rectangle class"<<endl;
		}
};
int main()
{
	shape *s;	
	shape s1;
	s=&s1;
	s->draw();
	rectangle r1;
	s=&r1;
	s->draw();
	return 0;
}

File: OOPS/runtime_polymorphisum.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace runtime_polymorphisum
{
    public class Shape // base class
    {
        public virtual void draw()  
        {
            Console.WriteLine("Draw...");
        }
    }

    public class Rectangle: Shape // calling base class
    {
        public override void draw() // overriding
        {
            Console.WriteLine("Draw Rectangle...");
        }
    }

    public class Circle: Shape // calling base class
    {
        public override void draw()  // overriding
        {
            Console.WriteLine("Draw Circle...");
        }
    }
    internal class Program
    {
        static void Main(string[] args)
        {
            Shape s; // base class Shape

            s = new Shape();
            s.draw();

            s = new Rectangle();
            s.draw();

            s = new Circle();
            s.draw();

            Console.ReadKey();
        }
    }
}
